consensus/src/ordered_broadcast/mod.rs

```diff
@@ -35,36 +35,46 @@
 
 use commonware_utils::Array;
 
-mod ack_manager;
-use ack_manager::AckManager;
-mod config;
-pub use config::Config;
-mod engine;
-pub use engine::Engine;
-mod ingress;
-use ingress::{Mailbox, Message};
-mod metrics;
 mod namespace;
 mod parsed;
 mod prover;
 pub use prover::Prover;
 mod serializer;
-mod tip_manager;
-use tip_manager::TipManager;
 mod wire {
     include!(concat!(env!("OUT_DIR"), "/wire.rs"));
 }
+
+cfg_if::cfg_if! {
+    if #[cfg(not(target_arch = "wasm32"))] {
+        mod ack_manager;
+        use ack_manager::AckManager;
+        mod config;
+        pub use config::Config;
+        mod engine;
+        pub use engine::Engine;
+        mod metrics;
+        mod tip_manager;
+        use tip_manager::TipManager;
+    }
+}
+
 #[cfg(test)]
 pub mod mocks;
 
-/// Used as the [`Index`](crate::Coordinator::Index) type.
-/// Defines the current set of sequencers and signers.
+/// Used as the [`Index`](crate::Supervisor::Index) type.
+/// Defines the current set of sequencers and validators.
 ///
 /// This is not a single "View" in the sense of a consensus protocol, but rather a continuous
-/// sequence of views in-which the set of sequencers and signers is constant.
+/// sequence of views in-which the set of sequencers and validators is constant.
 pub type Epoch = u64;
 
-/// Used as the [`Application::Context`](crate::Application::Context) type.
+/// Returns the current epoch to the [`Engine`].
+pub trait Epocher: Clone + Send + 'static {
```

Comment:
I think we may have some notion of this at the top-level trait eventually but think it is too early to say what it will look like exactly (and this is the best place for it in the meantime).